### PR TITLE
Fix bug re timestamps not being persisted: add timestamps to crdtentity...

### PR DESCRIPTION
**Pull request by galganif**
… as that is what get passed to the database for storage, and test to confirm they now come back.

### CODE REVIEW

File: java/arcs/core/crdt/CrdtEntity.kt [hunk 1]
```diff
@@ -80,6 +80,35 @@ class CrdtEntity(
...
+                // Two different values, we keep the earlier creationTimestamp.
+                _data.creationTimestamp = minOf(_data.creationTimestamp, other.creationTimestamp)
+            }
+        }
+        if (_data.expirationTimestamp != other.expirationTimestamp) {
+            allOps = false
+            if (_data.expirationTimestamp == RawEntity.UNINITIALIZED_TIMESTAMP) {
+                _data.expirationTimestamp = other.expirationTimestamp
+            } else if (other.expirationTimestamp != RawEntity.UNINITIALIZED_TIMESTAMP) {
+                // Two different values, we don't how to merge this.
```
Comment (mmandlis) [inline java/arcs/core/crdt/CrdtEntity.kt]: i think we should use the expiration timestamp of the same data we used creation timestamp of.
and in general, data can't have expiration if it doesn't have creation (i don't think we ever check for it, but it shouldn't be possible).

Comment (galganif) [reply java/arcs/core/crdt/CrdtEntity.kt]: after our discussion, I think throwing here is ok, and we should also throw for creationTimestamps, and these timestamps should be created with the id and then never change. I added a TODO for that as it doesn't work at the moment.

Comment (shans) [reply java/arcs/core/crdt/CrdtEntity.kt]: agreed, these should all be tightly tied together.

Comment (galganif) [reply java/arcs/core/crdt/CrdtEntity.kt]: I will work on the TODO asap.

**Pull request merged**